src/Action/OrderRefundsListAction.php

```diff
@@ -25,8 +30,24 @@ public function __construct(OrderRepositoryInterface $orderRepository, Environme
 
     public function __invoke(Request $request): Response
     {
+        /** @var OrderInterface $order */
         $order = $this->orderRepository->findOneByNumber($request->attributes->get('orderNumber'));
 
+        if ($order->getPaymentState() !== OrderPaymentStates::STATE_PAID) {
```

Comment:
I wonder if this check is really necessary... Of course, it protects from accessing refunds list page for not paid order... one the other hand we already do not display the **refunds** button and even if somebody would go there by URL, their still would not be possible to refund anything, as we have also check it in the command handler... It just adds big complexity to this action for a relatively small benefit ⚖️

Reply:
Or if we really need to check it, we can consider protecting it on the level of the security (by [expression](http://symfony.com/doc/current/security/access_control.html#securing-by-an-expression) or [voter](http://symfony.com/doc/current/security/voters.html))

Reply:
Well, for me it still seems like a security, admin should not have **access** to order's refunds page 😄 

Nevertheless, we should for sure extract this to some service like `OrderRefundsListGuard`... Here it should be only calling a service and redirecting to referer if it fails/throws an exception.

Reply:
ping @GSadee still relevant 😄